code/game/machinery/modular_shield.dm

```diff
@@ -0,0 +1,681 @@
+	current_regeneration = (max_regeneration / (0.5 + (radius * 2)/max_radius))
+
+	if(!exterior_only)
+		current_regeneration *= 0.5
+
+///Reduces the strength of the shield based on the given integer
+/obj/machinery/modular_shield_generator/proc/shield_drain(damage_amount)
+	stored_strength -= damage_amount
+	begin_processing()
+	if (stored_strength < 5)
+		recovering = TRUE
+		deactivate_shields()
+		stored_strength = 0
+		update_icon_state()
+
+/obj/machinery/modular_shield_generator/process(seconds_per_tick)
+	stored_strength = min((stored_strength + (current_regeneration * seconds_per_tick)),max_strength)
+	if(stored_strength == max_strength)
+		if (recovering)
+			recovering = FALSE
+			calculate_regeneration()
+			update_icon_state()
+		end_processing() //we dont care about continuing to update the alpha, we want to show history of damage to show its unstable
+	if (active)
+		var/random_num = rand(1,deployed_shields.len)
+		var/obj/structure/emergency_shield/modular/random_shield = deployed_shields[random_num]
+		random_shield.alpha = max(255 * (stored_strength/max_strength), 40)
+
+
+
+//Start of other machines
+///The general code used for machines that want to connect to the network
+/obj/machinery/modular_shield/module
+
+	name = "Modular Shield Debugger" //Filler name and sprite for testing
+	desc = "This is filler for testing you shouldn`t see this."
+	icon = 'icons/mecha/mech_bay.dmi'
+	icon_state = "recharge_port"
+	density = TRUE
+
+	///The shield generator we are connected to if we find one or a node provides us one
+	var/obj/machinery/modular_shield_generator/shield_generator
+
+	///The node we are connected to if we find one
+	var/obj/machinery/modular_shield/module/node/connected_node
+
+	///This is the turf that we are facing and able to search for connections through
+	var/turf/connected_turf
+
+/obj/machinery/modular_shield/module/Initialize(mapload)
+	. = ..()
+
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/Destroy()
+
+	if(shield_generator)
+		LAZYREMOVE(shield_generator.connected_modules, (src))
+		shield_generator.calculate_boost()
+	if(connected_node)
+		LAZYREMOVE(connected_node.connected_through_us, (src))
+	return ..()
+
+/obj/machinery/modular_shield/module/examine(mob/user)
+	. = ..()
+
+	if(isnull(shield_generator) && isnull(connected_node))
+		. += "It can be loosened and rotated with a screwdriver and wrench. It can be connected to a node or generator with a multitool."
+		return
+	. += "It can be loosed and rotated with a screwdriver and wrench, rotating it will sever its connection."
+
+/obj/machinery/modular_shield/module/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	panel_open = !(panel_open)
+	tool.play_tool_sound(src, 50)
+	update_icon_state()
+	if(panel_open)
+		to_chat(user, span_notice("You open the maintenance hatch of [src]."))
+		return TRUE
+	to_chat(user, span_notice("You close the maintenance hatch of [src]."))
```

Comment:
Should these be balloon alert?

Reply:
I tried to match the normal messages you get when screwdrivering other machines, are we moving away from that?

Reply:
done